### PR TITLE
[`flake8-bugbear`] Add `flake8-bugbear`'s B030 rule

### PR DESCRIPTION
**Pull request by aacunningham**
Addresses B030 in #2954 

Gave the rule name the good ol' college try, let me know if there's a better name we can think of.

This implementation is somewhat faithful to the original, with the exception (pun) that we'll handle iterables a little more strictly.

bugbear currently allows this:
```python
try:
    raise ValueError()
except (ValueError, (RuntimeError, (KeyError, TypeError))):
    pass
```
which causes a TypeError when `(RuntimeError, (KeyError, TypeError))` doesn't inherit from BaseException.

ruff will mark that as a failure, and will allow the following:
```python
try:
    raise ValueError()
except (ValueError, *(RuntimeError, *(KeyError, TypeError))): # Now with more stars
    pass
```

This seems like their intention, and I've already made a [PR to their project](https://github.com/PyCQA/flake8-bugbear/pull/364). Not sure how important it is that the rules match the base projects _exactly_, but we could wait and see how they respond to the change.

### CODE REVIEW

**Comment by charliermarsh** [comment]
Awesome. Agree w/ the change you made (and looks like upstream agrees too).

File: crates/ruff/src/rules/flake8_bugbear/rules/except_with_non_exception_classes.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::types::Range;
+
+use crate::checkers::ast::Checker;
+use crate::registry::Diagnostic;
+use crate::violation::Violation;
+
+#[violation]
+pub struct ExceptWithNonExceptionClasses;
```
Comment (charliermarsh) [inline crates/ruff/src/rules/flake8_bugbear/rules/except_with_non_exception_classes.rs:13]: I think the name is 👍

File: crates/ruff/src/rules/flake8_bugbear/rules/except_with_non_exception_classes.rs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+impl Violation for ExceptWithNonExceptionClasses {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("`except` handlers should only be exception classes or tuples of exception classes")
```
Comment (charliermarsh) [inline crates/ruff/src/rules/flake8_bugbear/rules/except_with_non_exception_classes.rs:18]: Tweaked this to use backticks around except which differs from bugbear but it's more consistent with how we tend to format messages.

File: crates/ruff/src/rules/flake8_bugbear/rules/except_with_non_exception_classes.rs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    }
+    flattened_exprs
+}
+
+/// B030
+pub fn except_with_non_exception_classes(checker: &mut Checker, excepthandler: &Excepthandler) {
+    let ExcepthandlerKind::ExceptHandler { type_, .. } = &excepthandler.node;
+    let Some(type_) = type_ else {
+        return;
+    };
```
Comment (charliermarsh) [inline crates/ruff/src/rules/flake8_bugbear/rules/except_with_non_exception_classes.rs:50]: Changed this to use `let-else`.

**Comment by charliermarsh** [comment]
Great PR! Thanks!

**Comment by aacunningham** [comment]
Appreciate the updates to the code and the merge! 😄

**Pull request merged**